### PR TITLE
Defer asking for age in the Google OAuth sign up flow

### PR DESCRIPTION
**Pull request by joshlory**
Users created this way will be asked for their age when they first sign in.  +cc @poorvasingal

When I am on the sign up page, and I click "Sign in with Google" as a new user:

screen shot 2017-06-15 at 11 19 20 am

## Before:

screen shot 2017-06-15 at 11 20 40 am

## After:

screen shot 2017-06-15 at 11 17 57 am

### CODE REVIEW

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -332,7 +332,8 @@ def district_teachers(cohort = nil)
   validates :name, length: {within: 1..70}, allow_blank: true
   validates :name, no_utf8mb4: true
 
-  validates :age, presence: true, on: :create # only do this on create to avoid problems with existing users
+  is_google = proc {|user| user.provider == 'google_oauth2'}
+  validates :age, presence: true, on: :create, unless: is_google # only do this on create to avoid problems with existing users
```
Comment (ashercodeorg) [inline dashboard/app/models/user.rb:336]: nit: This line is over 100 characters. And the comment should probably be updated.

**Comment by ashercodeorg** [comment]
Actually, I revoke my LGTM. This means that the Display Name is captured from the oauth account. Given that we want to encourage users to not use full names (e.g., as pictured) and we want to offer the opportunity to specify gender, I'd want sign-off from Product on this.

**Comment by poorvasingal** [comment]
I'm not worried about the gender issue - that's the same deal w/ word + picture accounts. 

I'm a bit torn about the display name issue. I think it'll be weird to not use the same name that they have in Google Classrooms. Plus, any change they make to their name will get overridden whenever we do a manual sync (since we'd want to respect name fixes made inside Google Classrooms). So I'm personally fine with this...

**Comment by ashercodeorg** [comment]
LGTM, then, given @poorvasingal's reasoning.

Though I want to be certain I understand future Product requirements, namely that we are going to (on every Google Classroom sign-on), sync data from the Google Classroom account to our Code Studio account?

**Comment by poorvasingal** [comment]
Just talked to Josh offline. Apparently I'm wrong about that (that the name will re-sync each time). I'll need to talk to Alice / Marina and get their thoughts here too, but all the potential paths I can see us following would build on top of the current flow we're proposing. So I'm fine with going forward with this flow.

**Pull request merged**